packages/nx/src/native/cache/http_remote_cache.rs

```diff
@@ -0,0 +1,206 @@
+use std::{
+    env,
+    fs::{self},
+    path::Path,
+};
+
+use super::cache::CachedResult;
+use flate2::Compression;
+use reqwest::{header, Client, ClientBuilder};
+use tar::{Archive, Builder};
+use tracing::trace;
+
+#[napi]
+pub struct HttpRemoteCache {
+    client: Client,
+    url: String,
+}
+
+#[napi]
+impl HttpRemoteCache {
+    #[napi(constructor)]
+    pub fn new() -> Self {
+        let mut headers = header::HeaderMap::new();
+        let auth_token = env::var("NX_SELF_HOSTED_REMOTE_CACHE_ACCESS_TOKEN");
+        if let Ok(token) = auth_token {
+            headers.insert(
+                header::AUTHORIZATION,
+                header::HeaderValue::from_str(&format!("Basic {}", token)).unwrap(),
```

Comment:
We're keeping this as `Bearer`.
```suggestion
                header::HeaderValue::from_str(&format!("Bearer {}", token)).unwrap(),
```